src/rate_limit.rs

```diff
@@ -0,0 +1,42 @@
+//! Rate Limit interface
+
+use hyper::client::Connect;
+
+use {Future, Github};
+
+pub struct RateLimit<C: Clone + Connect> {
+    github: Github<C>,
+}
+
+impl<C: Clone + Connect> RateLimit<C> {
+    #[doc(hidden)]
+    pub fn new(github: Github<C>) -> Self {
+        Self { github }
+    }
+
+    /// https://developer.github.com/v3/rate_limit/#get-your-current-rate-limit-status
+    pub fn get(&self) -> Future<RateLimitStatus> {
+        self.github.get("/rate_limit")
+    }
+}
+
+// representations
+
+#[derive(Debug, Deserialize)]
+pub struct RateLimitStatus {
+    pub resources: RateLimitResourcesStatus
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RateLimitResourcesStatus {
+    pub core: RateLimitResourceStatus,
+    pub search: RateLimitResourceStatus,
+    pub graphql: RateLimitResourceStatus,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RateLimitResourceStatus {
+    pub limit: u32,
+    pub remaining: u32,
+    pub reset: u32 // ideally something like std::time::Duration
```

Comment:
We can always define a fn on impl of the same name it would maybe pair well with [this](https://github.com/softprops/hubcaps/blob/master/src/lib.rs#L455)